Fix/code clean up/enemy control

rameziophobia opened the pull request.

MohamedSaidSallam committed:
deleted out/ folder

MohamedSaidSallam committed:
deleted out/ folder

MohamedSaidSallam committed:
Reset .idea files

also removed white spaces from .gitignore

MohamedSaidSallam committed:
moved assets to one folder

rameziophobia committed:
fix typo in enemy path

rameziophobia committed:
Merge branch 'master' into fix/CodeCleanUp/main

rameziophobia committed:
updating from master

rameziophobia committed:
refactor gameViewUi, fix paths

rameziophobia committed:
add enemy projectileControl

add spawnRing

rameziophobia committed:
fix HealthBars not showing

rameziophobia committed:
Merge branch 'fix/CodeCleanUp/main' into feat/enemy-projectile-control

rameziophobia committed:
add enemy attack patterns

add EnemyProjectileControl Class
add ProjectileControl parent class
add 3 enemy patterns
    towards player
    ring1by1
    ring all around

rameziophobia committed:
editing comments

MohamedSaidSallam committed:
refactor: a lot

rameziophobia committed:
refactor healthBars

MohamedSaidSallam committed:
style: Constants naming and position

rameziophobia committed:
fix: enemy not despawning, projectile animation

add levelManager.remove enemy in enemy update

add projectile .animate abl el spawn

rameziophobia committed:
Merge branch 'feat/enemy-projectile-control' into fix/CodeCleanUp/enemy-control

rameziophobia committed:
fix: merge bugs

rameziophobia committed:
refactor: remove unnecessary code

remove emptying projArray in enemy control

rameziophobia committed:
refactor: remove comments & useless code

rameziophobia committed:
remove ProjectileControl class

rameziophobia committed:
uncapitalize package names

rameziophobia committed:
fix codecy issues

rameziophobia committed:
fix codecy issue

MohamedSaidSallam commented on src/model/Entity.java:
```diff
@@ -0,0 +1,16 @@
+
+public abstract class Entity extends GameObject {
+
+    protected float speed;//todo
```
this was supposed to force sub classes to have a speed value (maybe added it to the constructor or smth)

MohamedSaidSallam commented on src/model/GameObject.java:
```diff
@@ -4,54 +4,46 @@
-
-    public  double calcDistanceToSpawner(){
-        return Math.hypot(spawner.getX() - 0, spawner.getY() - 0);
+        return url.contains("animated");//todo X: -.-
```
-.-

MohamedSaidSallam commented on src/model/player/Player.java:
```diff
@@ -2,77 +2,89 @@
-    private Bars ShieldRectangle;
+    private static final long REGENERATION_TIME_CD_MS = 5000;
+
+    private long NEXT_REGEN_TIME = 0; // todo ?????????????????????????????????????????????????????????????????????
```
??????????????????????????
it wasn't used if i remember correctly

MohamedSaidSallam replied:
nvm

MohamedSaidSallam commented on src/model/projectiles/Projectile.java:
```diff
@@ -1,74 +1,100 @@
-    public double getScale() {
-        return scale;
+    public double getDamage() {
+        return projectileType.getDAMAGE() * (((scale - 1) * 2) + 1); //todo wot ?
```
wot

MohamedSaidSallam commented:
there's a lot of scattered todo btw @rameziophobia

MohamedSaidSallam committed:
style: general changes

- auto fix formatting
- remove unused import statment
- remove unused variable (NEXT_REGEN_TIME)

rameziophobia committed:
add projectile collision w/ player

add projectile collision w/ player
increase player max hp
